src/libraries/Common/src/System/Net/SocketProtocolSupportPal.Unix.cs

```diff
@@ -21,7 +20,11 @@ private static unsafe bool IsSupported(AddressFamily af)
             IntPtr socket = invalid;
             try
             {
+#if SYSTEM_NET_SOCKETS_DLL
                 Interop.Error result = Interop.Sys.Socket(af, SocketType.Dgram, 0, &socket);
+#else
+                Interop.Error result = Interop.Sys.Socket((int)af, (int)Internals.SocketType.Dgram, 0, &socket);
+#endif
```

Comment:
Can we just always use the #else?

Reply:
Probably. I had difficulty with building NameResolution assembly. Because it is used by Sockets it seems like it cannot reference socket code. I can use the Int in socket and cast probably on one place.

Reply:
updated,.